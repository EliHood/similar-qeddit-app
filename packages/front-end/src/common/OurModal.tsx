import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import { useTheme } from '@material-ui/core/styles';
import OurLink from './OurLink';

export default function OurModal(props) {
    const theme = useTheme();
    const fullScreen = useMediaQuery(theme.breakpoints.down('sm'));

    return (
        <Dialog fullScreen={fullScreen} open={props.open} onClose={props.handleClose} aria-labelledby="responsive-dialog-title">
            <DialogTitle id="responsive-dialog-title">You have to be signed in to do that</DialogTitle>
            <DialogContent>
                <DialogContentText>
                    <OurLink
                        style={{ color: '#333' }}
                        to={{
                            pathname: '/login',
                        }}
                        title="Sign In"
                    />
                </DialogContentText>
            </DialogContent>
            <DialogActions>
                <Button autoFocus onClick={props.handleClose} color="primary">
                    Close
                </Button>
            </DialogActions>
        </Dialog>
    );
}
