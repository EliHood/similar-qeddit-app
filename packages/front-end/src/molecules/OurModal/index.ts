import OurModal from './OurModal';

export default OurModal;