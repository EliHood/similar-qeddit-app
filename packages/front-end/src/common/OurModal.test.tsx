import React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Button } from '@material-ui/core';
import OurModal from './OurModal';

describe('Should render <OurModal/>', () => {
    let wrapper;
    const props = {
        handleClose: jest.fn(),
        open: jest.fn(),
    };
    beforeAll(() => {
        wrapper = shallow(<OurModal open />);
    });
    it('should render <OurModal/>', () => {
        expect(wrapper).toHaveLength(1);
    });
    it('should find button', () => {
        expect(wrapper).toMatchSnapshot();
    });
});
