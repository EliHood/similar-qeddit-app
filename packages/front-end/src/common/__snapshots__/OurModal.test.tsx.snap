// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render <OurModal/> should find button 1`] = `
<WithStyles(ForwardRef(Dialog))
  aria-labelledby="responsive-dialog-title"
  fullScreen={false}
  open={true}
>
  <WithStyles(ForwardRef(DialogTitle))
    id="responsive-dialog-title"
  >
    You have to be signed in to do that
  </WithStyles(ForwardRef(DialogTitle))>
  <WithStyles(ForwardRef(DialogContent))>
    <WithStyles(ForwardRef(DialogContentText))>
      <OurLink
        style={
          Object {
            "color": "#333",
          }
        }
        title="Sign In"
        to={
          Object {
            "pathname": "/login",
          }
        }
      />
    </WithStyles(ForwardRef(DialogContentText))>
  </WithStyles(ForwardRef(DialogContent))>
  <WithStyles(ForwardRef(DialogActions))>
    <WithStyles(ForwardRef(Button))
      autoFocus={true}
      color="primary"
    >
      Close
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(DialogActions))>
</WithStyles(ForwardRef(Dialog))>
`;
